### ISSUE TITLE
Create a comparison chart of all hash functions

### ISSUE THREAD

**Issue by EdOverflow** [crypto101/book#303]
Maybe include a table like this in the `Hash functions` chapter to give readers an overview:

| Hash Function | Message Size | Block Size | Word Size | Digest Size |
|---------------|--------------|------------|-----------|-------------|
| MD5           | Unlimited    | 512        | 32        | 128         |
| SHA-1         | < 2^64       | 512        | 32        | 160         |
| SHA-256       | < 2^64       | 512        | 32        | 256         |
| SHA-384       | < 2^128      | 1024       | 64        | 384         |

**Comment by lvh** [comment]
Hm. Interesting. Do you have a specific reason why you'd think that'd be valuable?

(I'm a little touchy about this, not because I think it's.a bad idea; but when this recently made HN a specific criticism was that Crypto 101 is a crypto zoo rather than a guide. I think that's a mistaken impression from just looking at the TOC, but I want to make sure it stays that way.)

**Comment by EdOverflow** [comment]
I think it would be nice to explain all the terms listed in the table above. Then at the end of the chapter you could sum everything up with a general overview in form of a table.

About the HN comment: I don't agree with that statement since one can be guided through a zoo. On top of that, according to what I understand about the user's comment, wouldn't the same apply to Bruce Schneier's _Applied Cryptography_?

**Comment by lvh** [comment]
Yeah, I think you're right, and I don't want to be too distracted by _that guy_, but it's important that he stays wrong because that is explicitly not a goal :) Explaining all of the terms there is unambiguously a good idea, and, well, the table is small enough not to hurt, right?

**Comment by EdOverflow** [comment]
I agree with you. :)

**Comment by EdOverflow** [comment]
I created a similar table for the SHA-2 chapter here: #309.
